File: app/src/main/java/com/asafvaron/themoviedbtest/ui/movie_details/MovieDetailContract.java
package com.asafvaron.themoviedbtest.ui.movie_details;

import android.content.Context;

import com.asafvaron.themoviedbtest.model.Movie;

/**
 * Created by asafvaron on 19/03/2017.
 */

interface MovieDetailContract {
    interface View {
        void setMovieRunTime(int runTime);

        void failedToGetMovieRunTime(String err);
    }

    interface Presenter {
        void getMovieRunTime();

        void updateDb(Context context, Movie movie);
    }
}

File: app/src/main/java/com/asafvaron/themoviedbtest/ui/movie_details/MovieDetailContract.kt
package com.asafvaron.themoviedbtest.ui.movie_details

import android.content.Context

import com.asafvaron.themoviedbtest.model.Movie
import com.asafvaron.themoviedbtest.ui.BaseView

/**
 * Created by asafvaron on 19/03/2017.
 */

internal interface MovieDetailContract {
    interface View : BaseView<Presenter> {
        fun setMovieRunTime(runTime: Int)

        fun failedToGetMovieRunTime(err: String)
    }

    interface Presenter {
        fun getMovieRunTime()

        fun updateDb(context: Context, movie: Movie)
    }
}
